src/components/settings/Username.svelte

```diff
@@ -1,8 +1,34 @@
 <script lang="ts">
+  import clipboardCopy from 'clipboard-copy'
+
   import { sessionStore } from '$src/stores'
+  import { addNotification } from '$lib/notifications'
+  import ClipboardIcon from '$components/icons/ClipboardIcon.svelte'
+
+  const handleCopyUsername = async (): Promise<void> => {
+    await clipboardCopy($sessionStore.username)
+    addNotification('Copied to clipboard', 'success')
+  }
+
+  $: usernameParts = $sessionStore.username.split('#')
 </script>
 
 <div>
   <h3 class="text-lg mb-4">Username</h3>
-  <p>{$sessionStore.username}</p>
+  <div class="flex items-center">
+    <p>
+      {usernameParts[0]}
+      <span class="text-neutral-500 -ml-[3px]">
+        #{usernameParts[1].substring(0, 12)}...{usernameParts[1].substring(
+          usernameParts[1].length - 5
+        )}
```

Comment:
>Could even store this as fullTruncated in the username hash.

true! i think adding all these parts to the `username` object may be a bit confusing at this point since this is the only place they're used, but it's definitely something to keep in mind moving forward 👍🏼

Reply:
actually, i'll just make a util function for it 👍🏼

Reply:
I ended up making a `TruncatedUsername` component